### PR TITLE
Child Account Functionality

### PR DESCRIPTION
**Pull request by Tvcz**
Worked with @mayazeldin to create a new route to handle parents adopting sites for a specified child and a new route to handle parents recording stewardship activities for a specified child.

Route to handle parent adopt site for child:
`POST api/v1/protected/sites/:site_id/parent_adopt`

Route to handle parent record stewardship for child:
`POST api/v1/protected/sites/:site_id/parent_record_stewardship`

Created additional helper methods (`checkParent`, `isParent`) for the `ProtectedSiteProcessor` for determining whether a user is the parent of a given child.

[ClickUp Ticket](https://app.clickup.com/t/29xd1x4)

### CODE REVIEW

File: api/src/main/java/com/codeforcommunity/dto/site/RecordStewardshipRequest.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public RecordStewardshipRequest(
     this.weeded = weeded;
   }
 
-  private RecordStewardshipRequest() {}
+  protected RecordStewardshipRequest() {}
```
Comment (priime0) [inline api/src/main/java/com/codeforcommunity/dto/site/RecordStewardshipRequest.java:27]: nit: Maybe `default` would be better here instead?

File: api/src/main/java/com/codeforcommunity/dto/site/ParentRecordStewardshipRequest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.sql.Date;
+import java.util.List;
+
+public class ParentRecordStewardshipRequest extends RecordStewardshipRequest {
+
+  private Integer childUserId;
+
+  public ParentRecordStewardshipRequest(
+      Date date,
+      boolean watered,
```
Comment (priime0) [inline api/src/main/java/com/codeforcommunity/dto/site/ParentRecordStewardshipRequest.java:14]: discussion: Maybe in the future we should put these activities (`watered`, `mulched`, `cleaned`, `weeded`) in an object. Pinging @chromium-52 . Might take significant refactoring though.

**Review by priime0 — COMMENTED** [review]
lgtm :+1:

**Review by priime0 — APPROVED** [review]

File: api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java [hunk 3]
```diff
@@ -23,13 +25,22 @@ public interface IProtectedSiteProcessor {
   /** Removes the record in the adopted sites table linking the site to its current adopter */
   void forceUnadoptSite(JWTData userData, int siteId);
 
+  /** Adopts a tree for a child account through the parent account **/
```
Comment (rymaju) [inline api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java:28]: Are these purpose statements doing more than what the function name already provides?

File: api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java [hunk 4]
```diff
@@ -23,13 +25,22 @@ public interface IProtectedSiteProcessor {
...
+                       Date dateAdopted);
+
   /** Get users adopted sites */
   AdoptedSitesResponse getAdoptedSites(JWTData userData);
 
   /** Records a new stewardship activity in the stewardship table linked to the given site */
   void recordStewardship(
       JWTData userData, int siteId, RecordStewardshipRequest recordStewardshipRequest);
 
+  /** Records a new stewardship activity for a child account through a parent account */
```
Comment (rymaju) [inline api/src/main/java/com/codeforcommunity/api/IProtectedSiteProcessor.java:39]: Same as above

File: api/src/main/java/com/codeforcommunity/dto/site/ParentRecordStewardshipRequest.java (hunk 2 above)
Comment (rymaju) [reply api/src/main/java/com/codeforcommunity/dto/site/ParentRecordStewardshipRequest.java:14]: +1
iirc we do this to mirror how we eventually store this in the DB. Remember our persistence model need not be our program model.

File: api/src/main/java/com/codeforcommunity/dto/site/RecordStewardshipRequest.java (hunk 1 above)
Comment (rymaju) [reply api/src/main/java/com/codeforcommunity/dto/site/RecordStewardshipRequest.java:27]: We should throw more keywords at it until it works like good Java programmers. /s

**Review by rymaju — APPROVED** [review]
I didnt realize our website was so popular with children! Does SFTT get used in schools or something? 🤔

File: api/src/main/java/com/codeforcommunity/dto/site/ParentRecordStewardshipRequest.java (hunk 2 above)
Comment (priime0) [reply api/src/main/java/com/codeforcommunity/dto/site/ParentRecordStewardshipRequest.java:14]: That does make sense.

**Pull request merged**